### PR TITLE
Report calculated entropies

### PR DESCRIPTION
**Pull request by e3krisztian**
Extended existing code, so that entropy can be reported for outside use.

Refactor:
- shuffled around code to do only one thing in entropy-calculation functions

Fixes:
- typing for alternative imports from `unblob.math`
- logging of entropy charts, which was unfortunately broken due to global state in the `plotext` library

### CODE REVIEW

File: unblob/processing.py [hunk 1]
```diff
@@ -345,15 +347,26 @@ def _process_chunks(
 
         for chunk in unknown_chunks:
             carved_unknown_path = carve_unknown_chunk(self.carve_dir, file, chunk)
-            self._calculate_entropy(carved_unknown_path)
-            self.result.add_report(chunk.as_report())
+            entropy = self._calculate_entropy(carved_unknown_path)
+            self.result.add_report(chunk.as_report(entropy=entropy))
```
Comment (qkaiser) [inline unblob/processing.py:351]: I don't understand why you add the `EntropyReport` to the results in two different ways. One line 336-337 you check if entropy is set and then add it as is, on this line you don't check if it is set and add it using `chunk.as_report()`.

File: unblob/math.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
+from typing import Callable
+
+shannon_entropy: Callable[[bytes], float]
```
Comment (qkaiser) [inline unblob/math.py:3]: Out of curiosity, was the lack of typing reported by ruff or causing issues ? What got you to write this ?

File: tests/test_processing.py [hunk 3]
```diff
@@ -129,32 +129,34 @@ def test_calculate_buffer_size(
     )
 
 
-def test_draw_entropy_plot_error():
+def test_format_entropy_plot_error():
     with pytest.raises(TypeError):
-        draw_entropy_plot([])
+        format_entropy_plot(percentages=[], buffer_size=1024)
```
Comment (qkaiser) [inline tests/test_processing.py:134]: I would also add a test case where we calculate the entropy of an empty file.

```python
p = Path("/tmp/empty")
p.touch()
report = calculate_entropy(p)
```

File: unblob/processing.py (hunk 1 above)
Comment (e3krisztian) [reply unblob/processing.py:351]: yes, because the previous instance is not chunk related, the entropy report is added to the result directly *if* it was calculated, while in the current case the entropy report is part of the chunk-report (or not), but the chunk report can be added unconditionally.

File: unblob/math.py (hunk 2 above)
Comment (e3krisztian) [reply unblob/math.py:3]: This is older than ruff, it is pyright LSP (editor only problem), it has bothered me for a year.

This is how importing in `processing` looks to  me without it:
shannon_entropy-pyright

File: tests/test_processing.py (hunk 3 above)
Comment (e3krisztian) [reply tests/test_processing.py:134]: I am not sure about testing exactly this bit (in fact I wanted to delete this test as well :)), as it looks a very corner-case, which is already handled by existing code at https://github.com/onekey-sec/unblob/blob/a2c396fdd765006336bade6648aa9e41a6ad7847/unblob/processing.py#L275 but some test will be added to check, that entropy is reported when processing a file.

Comment (qkaiser) [reply tests/test_processing.py:134]: Makes sense.

**Review by qkaiser — APPROVED** [review]

**Comment by e3krisztian** [comment]
Since the approval a test has been added for reporting entropy for unknown chunks and for files, that are not extracted further - except for empty files.
And also rebased to main.
